### PR TITLE
feat: Add warning for Snap installations

### LINKED ISSUE #398: Detect and warn about Snap installs
Pressure Vessel and Snap do not appear to like each other so NSProton is completely broken.

This can easily be fixed by just using the official steam.deb, but users may think using the Ubuntu Store is the right way to go.

With this I propose adding detection for snap installs and warning the user about possible problems (and maybe adding information on the wiki to educate the user)

### PR DESCRIPTION
**Pull request by Jan200101**
Needs testing

closes #398

### CODE REVIEW

File: src-tauri/src/northstar/install.rs [hunk 1]
```diff
@@ -163,6 +163,22 @@ pub fn find_game_install_location() -> Result<GameInstall, String> {
     // Attempt parsing Steam library directly
     match steamlocate::SteamDir::locate() {
         Some(mut steamdir) => {
+            #[cfg(not(target_os = "linux"))]
```
Comment (Jan200101) [inline src-tauri/src/northstar/install.rs:166]: I fixed this check up, but otherwise all is good.

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -68,6 +68,9 @@ open = "3.2.0"
 winapi = "0.3.9"
 semver = "1.0"
 
+[target.'cfg(not(target_os="windows"))'.dependencies]
```
Comment (Jan200101) [inline src-tauri/Cargo.toml:71]: should be updated to be linux, not not windows

**Comment by Jan200101** [comment]
Rebasing to resolve conflicts and clean up the history.
No review existed and the prior state was broken so I force pushed.
Future changes will be additional commits.

File: src-tauri/src/northstar/install.rs [hunk 3]
```diff
@@ -168,6 +168,22 @@ pub fn find_game_install_location() -> Result<GameInstall, String> {
...
+                    Ok(snap_dir) => std::path::PathBuf::from(snap_dir),
+                    Err(_) => match dirs::home_dir() {
+                        Some(path) => path,
+                        None => std::path::PathBuf::new(),
+                    }
+                    .join("snap"),
+                };
+
+                if steamdir.path.starts_with(snap_dir) {
+                    log::warn!("Found Steam installed via Snap, you may encounter issues");
```
Comment (GeckoEidechse) [inline src-tauri/src/northstar/install.rs:183]: Just to be clear that this warning is not shown to end-user unless they run from CLI. I guess that's fine?

At least I don't know an easy way to propagate message back to UI. From the top of my head if we just return an `Error` it gets caugh somewhere higher up in the stack and just converted to a generic "Couldn't find install".

Might need to rework the error system to use custom error types but that's a whole other beast ^^"

Comment (Jan200101) [reply src-tauri/src/northstar/install.rs:183]: I think returning an Error is a bit much.
This isn't a fatal error but known to cause problems.

Maybe the the dialogs from #418 would be good for this?
Open a dialog, ask if the user wants to launch via snap regardless of any issues and continue

Comment (GeckoEidechse) [reply src-tauri/src/northstar/install.rs:183]: Yeah, could be an idea. I'd say we me merge as is, and then add dialog here once the other PR is merged.

Comment (Jan200101) [reply src-tauri/src/northstar/install.rs:183]: Sounds good

**Comment by GeckoEidechse** [comment]
As for testing, I guess we'll just do it in production ¯\\\_(ツ)\_/¯

**Comment by Jan200101** [comment]
I've tested it by manually injecting values on an Ubuntu LTS Desktop install.

steamlocate-rs does not currently support snap, https://github.com/WilliamVenner/steamlocate-rs/pull/30

**Comment by GeckoEidechse** [comment]
Welp, guess we just future proofed then a bit, haha

**Pull request merged**